library/core/src/hint.rs

```diff
@@ -382,6 +391,85 @@ pub fn spin_loop() {
 ///
 /// This makes our benchmark much more realistic to how the function would actually be used, where
 /// arguments are usually not known at compile time and the result is used in some way.
+///
+/// # How to use this
+///
+/// In practice, `black_box` serves two purposes:
+///
+/// 1. It prevents the compiler from making optimizations related to the value of the returned
+///    type
+/// 2. It forces the input to be calculated, even if its results are never used
```

Comment:
```suggestion
/// 1. It prevents the compiler from making optimizations related to the value returned by `black_box`.
/// 2. It forces the value passed to `black_box` to be calculated, even the return value of `black_box` is unused.
```

Wrap or tweak at your discretion, but your original wording just says "the value" and it's a bit unclear exactly what value is being discussed.

Reply:
```suggestion
/// 1. It prevents the compiler from making optimizations related to the value returned by `black_box`
/// 2. It forces the value passed to `black_box` to be calculated, even if the return value of `black_box` is unused
```